File: python/passger.py
import random
from encoding import passwordSymbols

def passGer(length, sym=True): # in: int/bool, out: str
	out = ''
	mode = 4 if sym else 3
	length = [i for i in range(length)]
	random.shuffle(length)
	for i in length:
		if (i % mode == 0):
			out += passwordSymbols[0][random.randint(0, len(passwordSymbols[0]) - 1)]
		elif (i % mode == 1):
			out += passwordSymbols[1][random.randint(0, len(passwordSymbols[1]) - 1)]
		elif (i % mode == 2):
			out += passwordSymbols[2][random.randint(0, len(passwordSymbols[2]) - 1)]
		else:
			out += passwordSymbols[3][random.randint(0, len(passwordSymbols[3]) - 1)]
	return out

File: ts/src/passger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.passGer = passGer;
const encoding_1 = require("./encoding");
function passGer(len = 20, symbol = true) {
    let temp = [...Array(len).keys()], out = "", mode = symbol ? 4 : 3;
    shuffle(temp);
    temp.forEach((item) => {
        switch (item % mode) {
            case 0:
                out +=
                    encoding_1.passwordSymbols[0][randomInteger(0, encoding_1.passwordSymbols[0].length - 1)];
                break;
            case 1:
                out +=
                    encoding_1.passwordSymbols[1][randomInteger(0, encoding_1.passwordSymbols[1].length - 1)];
                break;
            case 2:
                out +=
                    encoding_1.passwordSymbols[2][randomInteger(0, encoding_1.passwordSymbols[2].length - 1)];
                break;
            case 3:
                out +=
                    encoding_1.passwordSymbols[3][randomInteger(0, encoding_1.passwordSymbols[3].length - 1)];
                break;
        }
    });
    return out;
}
function shuffle(array) {
    for (let i = array.length - 1; i > 0; i--) {
        const j = Math.floor(Math.random() * (i + 1));
        [array[i], array[j]] = [array[j], array[i]];
    }
}
function randomInteger(min, max) {
    let rand = min + Math.random() * (max + 1 - min);
    return Math.floor(rand);
}
